### DISCUSSION TITLE
How to destructure types

### DISCUSSION THREAD

**Discussion by wilkomanger** [gleam-lang/gleam#2255] [Questions & support] [answered]
Perhaps I've missed it, but I saw that you can destructure lists and tuples: (Missing from the Language Tour btw)
```gleam
let [x, y] = [1, 2]
let #(u, v) = $(3, 4)
```
How can this be done with types? Such as:
```gleam
type Something {
    Something(first: Int, second: Int)
}

let (first, second) = function_that_returns_something()
```

**Comment by wilkomanger** [comment, marked as answer]
Figured it out :smile: 
```gleam
type Something {
    Something(first: Int, second: Int)
}

let Something(first, second) = function_that_returns_something()
```
Would be nice to mention in the [Let bindings page in the Language Tour](https://gleam.run/book/tour/let-bindings.html)!

**Reply by lpil** [reply]
It's in the custom types section rather than the let section as let is introduced first, and the reader has to know about both :)

https://gleam.run/book/tour/custom-types.html

**Reply by wilkomanger** [reply]
Yeah makes sense!
